todoist/managers/archive.py

```diff
@@ -0,0 +1,188 @@
+"""
+Managers to get the list of archived items and sections.
+
+Manager makers available as "items_archive" and "sections_archive" attributes of
+API object.
+
+
+Usage example (for items).
+
+```python
+
+# Create an API object
+import todoist
+api = todoist.TodoistAPI(...)
+
+# Get project ID (take inbox)
+project_id = api.user.get()['inbox_project']
+
+# Initiate ItemsArchiveManager
+archive = api.items_archive.for_project(project_id)
+
+# Iterate over the list of completed items for the archive
+for item in archive.items():
+    print(item["date_completed"], item["content"])
+```
+"""
+
+from typing import TYPE_CHECKING
+
+from ..models import Item, Model, Section
+from .generic import Manager
+
+if TYPE_CHECKING:
+    from ..api import TodoistAPI
+
+
+class ArchiveManager(Manager):
+
+    object_model = Model
+
+    def __init__(self, api, object_type):
+        # type: (TodoistAPI, str) -> None
+        assert object_type in {"sections", "items"}
+        self.api = api
```

Comment:
The [Manager](https://github.com/Doist/todoist-python/blob/master/todoist/managers/generic.py) class that this class is based on, already includes the `api` property and sets it during initialization: https://github.com/Doist/todoist-python/blob/14dc945004d5a39b7f621dc93ef9001a8f6c091e/todoist/managers/generic.py#L9

So this shouldn't be needed, unless I'm missing something.

Reply:
Addressed in 0647039ea79af38e58472991e0d0c7f1a8fe5fd8